app/Legacy/Legacy.php

```diff
@@ -187,4 +106,33 @@ public static function translateLegacyPhotoID(int $photoID, Request $request): ?
 	{
 		return self::translateLegacyID($photoID, 'photos', $request);
 	}
+
+	/**
+	 * Given a username, password and ip (for logging), try to log the user as admin.
+	 * Returns true if succeeded, false if failed.
+	 *
+	 * @param string $username
+	 * @param string $password
+	 * @param string $ip
+	 *
+	 * @return bool
+	 */
+	public static function logAsAdmin(string $username, string $password, string $ip): bool
```

Comment:
Why is this method in `Legacy`? And we should name it properly
```suggestion
	public static function loginAsAdmin(string $username, string $password, string $ip): bool
```